compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3895,9 +3939,35 @@ private BType validateOverlappingFields(BRecordType recordType, BField origField
         return symTable.semanticError;
     }
 
+    private void removeErrorFromCompoundTypes(List<BType> remainingTypes) {
```

Comment:
This only removes error from readonly right? Shall we update the name of the method to reflect this?